test/e2e/specs/heatmaps/00-heatmap.spec.js

```diff
@@ -0,0 +1,379 @@
+            });
+        });
+
+        it('hovering over link shows the tooltip with content "View Array Data related to this Gene"', function(){
+            browser.actions()
+                .mouseMove(viewArrLinkElem)
+                .perform();
+
+            browser.wait(EC.visibilityOf(viewArrLinkTooltipElem), browser.params.defaultTimeout);
+            viewArrLinkTooltipElem.getText().then(function(text){
+                expect(text).toBe(expectedValue.arrayDataTableLinkHoverTooltip);
+            });
+        });
+
+        it('4 heatmaps are visible', function(){
+            heatmapsElem.count().then(function(cnt){
+                expect(cnt).toBe(4);
+            });
+        });
+    });
+
+    /**
+     * [first heatmap]
+     * title should be `Developing Pelvic Ganglia FACS (ST1)
+     * legend should be visible
+     * legend should have values: `9.5`, `10`, `10.5`
+     * heatmap bar should have 15 sections
+     */
+
+    describe(appName + ' first heatmap', function(){
+        var heatmapUrl = baseUrl + testConfig.geneExpression.heatmapNCBIGeneID17740,
+            EC = protractor.ExpectedConditions,
+            spinnerElem,
+            heatmapElem,
+            heatmapsElem,
+            heatmapID = Object.keys(expectedValue.NCBIGeneID17740)[0];
+
+        beforeAll(function(){
+            // Elements
+            spinnerElem = element(by.id('heatmapSpinner'));
+            heatmapsElem = element.all(by.repeater('heatmap in heatmaps'));
+            browser.ignoreSynchronization = true;
+            browser.get(heatmapUrl);
+            browser.wait(EC.invisibilityOf(spinnerElem), browser.params.defaultTimeout);
+            
+            heatmapElem = heatmapsElem.filter(function(elem, index){
+                return elem.element(by.tagName("heatmap")).getAttribute("heatmap-id").then(function(id){
+                    return id == heatmapID;
+                })
+            });
+        });
+        
+        it('title should be "Developing Pelvic Ganglia FACS (ST1)"', function(){
+            heatmapElem
+            .first()
+            .element(by.css('.infolayer .g-gtitle'))
+            .getText()
+            .then(function(text){
+                expect(text).toBe(expectedValue.NCBIGeneID17740[heatmapID].title);
+            });
+        });
+
+        it('legend should be visible', function(){
+            heatmapElem
+            .first(function(elem){
+                expect(elem.element(by.css('.infolayer .colorbar')).isPresent()).toBeTruthy();
+            });
+        });
+
+        it('legend should have values: `9.5`, `10`, `10.5`', function(){
+            heatmapElem
+            .first()
+            .all(by.css(".infolayer .crisp text"))
+            .getText()
+            .then(function(textArr){
+                var legendValues = []
+                for( var i = 0; i < textArr.length; i++){
+                (function(index){
```

Comment:
You shouldn't need to define an anonymous function here. The for loop isn't creating any promises that would be executed later at a different time. If you are unsure of what I mean, I can go over it in more detail. 

There are some places in our existing tests where we loop have loops that surround `describe` blocks and other function calls that generate promises where capturing the value of the variable (index in most cases) is important because that function will be executed at some point later in time rather than right after the loop is called.